src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -513,6 +529,77 @@ public HashSet<Guid> Dependencies
             }
         }
 
+        /// <summary>
+        /// Gathers the packages that this graph depends on
+        /// </summary>
+        public IEnumerable<IPackage> PackageDependencies
+        {
+            get
+            {
+                // Get all packages installed locally
+                var localPackages = GetLocalPackages();
+                // Get set of assemblies used by nodes in this workspace
+                var assembliesUsed = GetAssembliesUsed();
+
+                // Create a dictionary that maps assembly names to the package they are contained in
+                var assemblyPackageDict = new Dictionary<string, IPackage>();
+                foreach(var package in localPackages)
+                {
+                    foreach(var assemblyName in package.AssemblyNames)
+                    {
+                        assemblyPackageDict[assemblyName.FullName] = package;
+                    }
+                }
+
+                // Create a list of packages that are used in this workspace
+                var packageDependencies = new HashSet<IPackage>();
+                foreach(var assemblyName in assembliesUsed)
+                {
+                    if (assemblyPackageDict.ContainsKey(assemblyName.FullName))
+                    {
+                        packageDependencies.Add(assemblyPackageDict[assemblyName.FullName]);
```

Comment:
What about custom node packages. Those won't have assemblies.

Reply:
Thanks for pointing this out. I just added support for this here: https://github.com/DynamoDS/Dynamo/pull/9723/commits/91b55c575f2026724582b7884917c32d6c980cad